### PR TITLE
Rename `ComponentGraph`'s `get_parents` to `get_inputs`

### PR DESCRIPTION
**Pull request by angela97lin**
Breaking up #2490 into smaller and cleaner PRs :)

This PR renames `ComponentGraph`'s `get_parents` to `get_inputs`; I think this is a little more descriptive of what the method does, especially now that we are requiring .x/.y to be explicitly stated.

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@ Release Notes
     * Enhancements
     * Fixes
     * Changes
+        * Renamed ``ComponentGraph``'s ``get_parents`` to ``get_inputs`` :pr:`2540`
```
Comment (dsherry) [inline docs/source/release_notes.rst:7]: Let's also add to breaking changes

**Review by dsherry — APPROVED** [review]
LGTM!

**Review by chukarsten — APPROVED** [review]
These are too easy, it feels like a trap.  Soon there's going to be a 200+ file changed PR, I know it.

File: evalml/pipelines/component_graph.py [hunk 2]
```diff
@@ -499,14 +499,14 @@ def get_estimators(self):
...
+    def get_inputs(self, component_name):
+        """Retrieves all inputs for a given component.
 
         Arguments:
-            component_name (str): Name of the child component to look up
+            component_name (str): Name of the component to look up.
 
         Returns:
-            list[str]: List of inputs to use
+            list[str]: List of inputs to for the component.
```
Comment (ParthivNaresh) [inline evalml/pipelines/component_graph.py:509]: Might have accidentally deleted `use` here?

**Review by ParthivNaresh — APPROVED** [review]
Looks solid, just a typo I think but otherwise good to go!

File: evalml/pipelines/component_graph.py (hunk 2 above)
Comment (angela97lin) [reply evalml/pipelines/component_graph.py]: oops, ty :P

**Comment by angela97lin** [comment]
@chukarsten Nonono, these PRs are so that https://github.com/alteryx/evalml/pull/2490 doesn't have to be a 1000+ line change PR 😭

**Pull request merged**